app/domain/authentication/authn_k8s/application_identity.rb

```diff
@@ -0,0 +1,181 @@
+module Authentication
+  module AuthnK8s
+
+    Err = Errors::Authentication::AuthnK8s
+    # Possible Errors Raised: MissingNamespaceConstraint, IllegalConstraintCombinations
+    # ScopeNotSupported, ArgumentError
+
+    # This class defines a application identity of a given conjur host.
+    # The constructor initializes an ApplicationIdentity object and validates that
+    # it is configured correctly.
+    # The difference between the validation in this constructor and
+    # the validation in ValidateApplicationIdentity is that here we validate that
+    # the application identity is configured correctly, and thus is a valid application
+    # identity. In ValidateApplicationIdentity we validate that the defined application
+    # identity is actually the correct one in kubernetes.
+    # For example, a application identity `some-namepsace/blah/some-value` is not
```

Comment:
NIT: "an application"

Reply:
thanks. missed this in the `machine` -> `application` refactor.